main/config.c

```diff
@@ -363,20 +392,65 @@ struct ast_config_include *ast_include_new(struct ast_config *conf, const char *
 	else
 		inc->included_file = ast_strdup(included_file);
 
-	inc->exec = is_exec;
-	if (is_exec)
-		inc->exec_file = ast_strdup(exec_file);
-
 	if (!inc->include_location_file
 		|| !inc->included_file
-		|| (is_exec && !inc->exec_file)) {
+		|| ((include_type == CONFIG_STATEMENT_EXEC) && !inc->exec_file)) {
```

Comment:
You moved the `inc->exec_file = ast_strdup(exec_file);` so that it occurs after this check. I haven't tested but I assume this means we won't get the `#exec` rendered in to the saved file.